components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/APIMappingUtil.java

```diff
@@ -590,9 +590,11 @@ public static APIDTO fromAPItoDTO(API model) throws APIManagementException {
         }
 
         //Get Swagger definition which has URL templates, scopes and resource details
-        String apiSwaggerDefinition = apiProvider.getOpenAPIDefinition(model.getId());
-        List<APIOperationsDTO> operationsDTOs = getOperationsFromSwaggerDef(model, apiSwaggerDefinition);
-        dto.setOperations(operationsDTOs);
+        if (!APIDTO.TypeEnum.WS.toString().equals(model.getType())) {
+            String apiSwaggerDefinition = apiProvider.getOpenAPIDefinition(model.getId());
+            List<APIOperationsDTO> operationsDTOs = getOperationsFromSwaggerDef(model, apiSwaggerDefinition);
+            dto.setOperations(operationsDTOs);
```

Comment:
Suggestion: 
dto.setOperations(getOperationsFromSwaggerDef(model, apiSwaggerDefinition));

Reply:
Having two lines might be easier for debugging purposes. Open for discussion :)

Reply:
Better to have two line IMO :)